src/pages/learn/introspection.mdx

```diff
@@ -146,25 +144,36 @@ tell us what these are lists of.
 }
 ```
 
-Let's end with a feature of the introspection system particularly useful
-for tooling; let's ask the system for documentation!
+Let's end with a feature of the introspection system particularly useful for tooling; let's ask the system for documentation:
 
 ```graphql
 # { "graphiql": true }
-{
+query {
   __type(name: "Droid") {
     name
     description
   }
 }
 ```
 
-So we can access the documentation about the type system using introspection,
-and create documentation browsers, or rich IDE experiences.
+As demonstrated above, we can access the documentation about the type system using introspection and create documentation browsers or rich IDE experiences.
+
+This has just scratched the surface of the introspection system; we can query for Enum type values, what Interface types another type implements, and more. We can even introspect on the introspection system itself.
+
+To see an example of a specification-compliant GraphQL query introspection system implemented in code, you can view the [introspection.ts](https://github.com/graphql/graphql-js/blob/main/src/type/introspection.ts) file in GraphQL.js repository.
+
+## Introspection in production
+
+Introspection is a useful feature of GraphQL, especially for client developers. However, for private GraphQL APIs, it may be undesirable to expose sensitive information about the schema publicly via the `__schema` field in a production environment.
+
+In those cases, it may be preferable to disable introspection in production environments as a part of a broader API security strategy that could also include authentication and authorization, depth-limiting, query cost analysis, and more.
+
+## Next steps
+
+To recap what we've learned about introspection:
+
+- Type names can be queried in a field selection set for an Object, Interface, or Union type using the `__typename` meta-field
+- Information about the elements of a GraphQL schema can be queried using the `__schema` field on the `query` root operation type
+- Introspection is often disabled in production environments for private APIs
```

Comment:
Even public APIs sometimes disable introspection and instead provide SDL files, and some who enable introspection in production have commented on how much effort it is to support! So let's simplify this:

```suggestion
- Introspection is often disabled in production environments
```